.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,147 @@
+name: Main pipeline
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+    inputs:
+      version_override:
+        description: 'Version Number'     
+        required: true
+        default: '0.0.0'
+
+jobs:
+  draftRelease:
+    name: Draft release
+    runs-on: ubuntu-latest
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v5
+        if: github.event_name == 'push'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  buildTest:
+    name: Build and test
+    runs-on: windows-latest 
+    needs: draftRelease
+    outputs:
+      buildNumber: ${{ steps.ver.outputs.BUILD_NUMBER }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: GetVersion
+        id: ver
+        shell: pwsh
+        run: |
+              if ("${{github.event.inputs.version_override}}")
+              {
+                Write-Host "Using manually entered override version"
+                $version = "${{ github.event.inputs.version_override}}"
+              }
+              else
+              {
+                $header = @{Authorization = 'Bearer ${{ secrets.GITHUB_TOKEN }}' }
+                $endPoint = "https://api.github.com/repos/${{github.repository}}/releases"
+                $info = Invoke-RestMethod -Uri $endPoint -Headers $header
+                $version = $info.name[0].Trim("v")
+              }
+              
+              Write-Host "::set-output name=BUILD_NUMBER::$($version)"
```

Comment:
Maybe a good idea to also `Write-Host` the version without the special `::set-output` syntax, so it gets logged in a more obvious spot than "buried in the Build and Analyze step's command" for debugging purposes?

Reply:
Added